core/src/main/java/org/ehcache/Ehcache.java

```diff
@@ -556,6 +568,7 @@ public void putAll(final Map<? extends K, ? extends V> entries) throws BulkCache
           resilienceStrategy.putAllFailure(entries, e, new BulkCacheWritingException(failures, successes));
         }
       }
+      markAllCacheEventsFailed();
```

Comment:
This will never be applied if line 554 is executed. Consider moving it to a `finally` block.